### PR TITLE
Code coverage

### LINKED ISSUE #16: Add code coverage to build
- nyc package

### PR DESCRIPTION
**Pull request by Joshua-Phelps**
closes #16 

- installed and configured nyc package to work with mocha and typescript
- I chose to use istanbul/nyc because it works well with mocha and has options to work with typescript

### CODE REVIEW

File: .mocharc.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+require: 'ts-node/register'
+full-trace: true
+bail: true
```
Comment (noelbundick) [inline .mocharc.yml:3]: Let's remove this line so we get all failures in CI

File: package.json [hunk 2]
```diff
@@ -19,13 +19,31 @@
...
+    ],
+    "require": [
+      "ts-node/register"
+    ],
+    "reporter": [
+      "text-summary",
+      "html"
+    ],
+    "sourceMap": true,
+    "instrument": true
```
Comment (noelbundick) [inline package.json:37]: Let's find out what this does or remove it

File: package.json [hunk 3]
```diff
@@ -19,13 +19,31 @@
...
+    ],
+    "extension": [
+      ".ts"
+    ],
+    "require": [
+      "ts-node/register"
+    ],
+    "reporter": [
+      "text-summary",
+      "html"
```
Comment (noelbundick) [inline package.json:34]: nit: do we need html output?

**Review by noelbundick — CHANGES_REQUESTED** [review]
looks good! Maybe a few tweaks - particularly for bailing on tests

File: .mocharc.yml (hunk 1 above)
Comment (noelbundick) [reply .mocharc.yml:3]:
```suggestion
```

File: package.json (hunk 2 above)
Comment (noelbundick) [inline package.json:37]:
```suggestion
    "sourceMap": true
```

**Review by Joshua-Phelps — COMMENTED** [review]
lgtm

**Review by noelbundick — APPROVED** [review]
lgtm

**Pull request merged**